File: integration-tests/java-selenium/java-selenium-test/src/main/java/com/gu/fronts/integration/test/page/Editions.java
package com.gu.fronts.integration.test.page;

import static com.gu.fronts.integration.test.page.util.PageElementHelper.elementIsALink;
import static java.lang.String.format;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.gu.fronts.integration.test.fw.selenium.ByTestAttributeSelector;
import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute;
import com.gu.fronts.integration.test.page.util.AbstractParentPage;

public class Editions extends AbstractParentPage {
    @FindByTestAttribute(using = "editions")
    private WebElement editions;

    @FindByTestAttribute(using = "edition-US")
    private WebElement editionUS;

    @FindByTestAttribute(using = "edition-UK")
    private WebElement editionUK;

    @FindByTestAttribute(using = "edition-AU")
    private WebElement editionAU;

    public Editions(WebDriver webDriver) {
        super(webDriver);
    }

    @Override
    public Editions isDisplayed() {
        super.exists(editions, editionUK, editionUS, editionAU);
        return this;
    }

    public NetworkFrontPage selectUSEdition() {
        editionUS.click();
        return pageFactory.initPage(webDriver, NetworkFrontPage.class);
    }

    public NetworkFrontPage selectUKEdition() {
        editionUK.click();
        return pageFactory.initPage(webDriver, NetworkFrontPage.class);
    }

    public NetworkFrontPage selectAUEdition() {
        editionAU.click();
        return pageFactory.initPage(webDriver, NetworkFrontPage.class);
    }

    public Editions usEditionSelected() {
        return editionSelected(editionUS);
    }

    public Editions isUkEditionSelected() {
        return editionSelected(editionUK);
    }

    public Editions auEditionSelected() {
        return editionSelected(editionAU);
    }

    private Editions editionSelected(WebElement editionElement) throws AssertionError {
        if (elementIsALink(editionElement)) {
            throw new AssertionError(format("%s was not selected",
                    editionElement.getAttribute(ByTestAttributeSelector.TEST_ATTR_NAME)));
        }
        return this;
    }

    public Editions usUsEditionPresent() {
        super.exists(editionUS);
        return this;
    }

    public Editions isAuEditionPresent() {
        super.exists(editionAU);
        return this;
    }

}

File: integration-tests/scala-selenium/app/com/gu/fronts/integration/test/page/common/Editions.scala
package com.gu.fronts.integration.test.page.common

import java.lang.String.format

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement

import com.gu.fronts.integration.test.fw.selenium.ByTestAttributeSelector
import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute
import com.gu.fronts.integration.test.page.nwfront.NetworkFrontPage
import com.gu.fronts.integration.test.page.util.FrontsParentPage
import com.gu.fronts.integration.test.page.util.PageElementHelper.elementIsALink
import com.gu.fronts.integration.test.page.util.PageElementHelper.existsAndDisplayed

class Editions(webDriver: WebDriver) extends FrontsParentPage(webDriver) {

  @FindByTestAttribute(using = "editions")
  private var editions: WebElement = _

  @FindByTestAttribute(using = "edition-US")
  private var editionUS: WebElement = _

  @FindByTestAttribute(using = "edition-UK")
  private var editionUK: WebElement = _

  @FindByTestAttribute(using = "edition-AU")
  private var editionAU: WebElement = _

  override def isDisplayed(): Boolean = {
    existsAndDisplayed(editions, editionUK, editionUS, editionAU)
  }

  def selectUSEdition(): NetworkFrontPage = {
    editionUS.click()
    pageFactory.initPage(webDriver, classOf[NetworkFrontPage])
  }

  def selectUKEdition(): NetworkFrontPage = {
    editionUK.click()
    pageFactory.initPage(webDriver, classOf[NetworkFrontPage])
  }

  def selectAUEdition(): NetworkFrontPage = {
    editionAU.click()
    pageFactory.initPage(webDriver, classOf[NetworkFrontPage])
  }

  def usEditionSelected(): Editions = editionSelected(editionUS)

  def isUkEditionSelected(): Editions = editionSelected(editionUK)

  def auEditionSelected(): Editions = editionSelected(editionAU)

  private def editionSelected(editionElement: WebElement): Editions = {
    if (elementIsALink(editionElement)) {
      throw new AssertionError(format("%s was not selected", editionElement.getAttribute(ByTestAttributeSelector.TEST_ATTR_NAME)))
    }
    this
  }

  def isUsEditionPresent(): Editions = {
    existsAndDisplayed(editionUS)
    this
  }

  def isAuEditionPresent(): Editions = {
    existsAndDisplayed(editionAU)
    this
  }
}